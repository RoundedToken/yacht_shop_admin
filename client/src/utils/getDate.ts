export function getDate(isoDate: string): string {
    const date = new Date(isoDate);

    return date.toLocaleString('default', {
        day: '2-digit',
        month: 'short',
        year: 'numeric',
        hour: 'numeric',
        minute: 'numeric',
        hour12: false,
    });
}
