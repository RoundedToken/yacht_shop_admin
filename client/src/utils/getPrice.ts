export const getPrice = (price: number) => {
    const formatter = new Intl.NumberFormat('et-EE', {
        style: 'currency',
        currency: 'EUR',
    });
    return formatter.format(price);
};
