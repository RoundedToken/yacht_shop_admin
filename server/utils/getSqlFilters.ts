import { IFilters } from '../types';

export const getSqlFilters = (filters: string, isProducts: boolean = false) => {
    const filtersObj = JSON.parse(filters) as IFilters;
    const productId = filtersObj.product;

    const filterStr =
        Object.entries(filtersObj)
            .filter((filter) => filter[1] !== null && filter[1] !== '')
            .map((filter) => {
                const key = filter[0];
                const value = filter[1];

                if (key === 'dateFrom') {
                    return `ord.DateEnter >= CAST('${value}T00:00:00' AS datetime)`;
                } else if (key === 'dateTo') {
                    return `ord.DateEnter <= CAST('${value}T23:59:59' AS datetime)`;
                } else if (key === 'status') {
                    return `ord.status IN (${value.join(',')})`;
                } else if (key === 'isPaid') {
                    return `ord.paid = ${value ? 1 : 0}`;
                } else if (key === 'product') {
                    if (isProducts) return '1=1';
                    else return `ordStr.tovar = ${value}`;
                } else if (key === 'client') {
                    return `ord.clientid = ${value}`;
                } else {
                    return `STUFF((SELECT DISTINCT ', ' + oplata.type  
                    FROM Oplata 
                    WHERE Oplata.orderid = ord.orderid  
                    FOR XML PATH('')), 1, 2, '') LIKE '%${value}%'`;
                }
            })
            .join(' AND ') || '1=1';

    return { filterStr, productId };
};
