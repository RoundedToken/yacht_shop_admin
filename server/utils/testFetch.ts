export class testFetch {
    private endpoint: string;

    constructor(endpoint: string) {
        this.endpoint = endpoint;
    }

    async getData(params: Record<string, string>) {
        const baseUrl = 'http://192.168.1.37:5000/api';
        const url = baseUrl + this.endpoint + '?' + new URLSearchParams(params);

        const response = await fetch(url);

        return await response.json();
    }
}
