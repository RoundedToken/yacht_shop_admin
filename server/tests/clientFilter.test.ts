import { test, describe } from 'vitest';
import { testFetch } from '../utils/testFetch';

describe.concurrent('Client Filter', () => {
    const getSales = new testFetch('/get_sales');
    const getProducts = new testFetch('/get_products');

    test('Get Sales client: 11226', async ({ expect }) => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":null,"product":null,"client":11226}',
        };
        const data = await getSales.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Products client: 11226', async ({ expect }) => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":null,"product":null,"client":11226}',
        };
        const data = await getProducts.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Sales client: 11243', async ({ expect }) => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":null,"product":null,"client":11243}',
        };
        const data = await getSales.getData(params);

        expect(data).toMatchSnapshot();
    });

    test('Get Products client: 11243', async ({ expect }) => {
        const params = {
            pageIndex: '0',
            pageSize: '10',
            sortId: 'Id',
            sortType: 'DESC',
            filters:
                '{"dateFrom":"2020-11-01","dateTo":"2022-10-04","payType":null,"status":[1,-1],"isPaid":null,"product":null,"client":11243}',
        };
        const data = await getProducts.getData(params);

        expect(data).toMatchSnapshot();
    });
});
