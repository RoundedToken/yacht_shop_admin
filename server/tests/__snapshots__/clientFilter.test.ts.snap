// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Filter Get Sales client: 11226 1`] = `
{
  "clients": [
    {
      "id": null,
      "name": null,
    },
    {
      "id": 245,
      "name": "Щербаков Антон Васильевич",
    },
    {
      "id": 627,
      "name": "Sailtech Oy (Финляндия)",
    },
    {
      "id": 676,
      "name": "Maritim-Finland",
    },
    {
      "id": 940,
      "name": "web",
    },
    {
      "id": 1108,
      "name": "TREM",
    },
    {
      "id": 1702,
      "name": "MarineTech (Germany)",
    },
    {
      "id": 2766,
      "name": "Optiparts",
    },
    {
      "id": 2773,
      "name": "Windex (Sweden)",
    },
    {
      "id": 9901,
      "name": "Anton Scherbakov",
    },
    {
      "id": 11182,
      "name": "_jamuuk",
    },
    {
      "id": 11183,
      "name": "PJK",
    },
    {
      "id": 11185,
      "name": "Regate (LV)",
    },
    {
      "id": 11186,
      "name": "TommyyLyy",
    },
    {
      "id": 11187,
      "name": "Pro Marine Trade",
    },
    {
      "id": 11188,
      "name": "Татьяна Каверина (Armastus)",
    },
    {
      "id": 11189,
      "name": "NORA LII SOPRADE SELTS",
    },
    {
      "id": 11190,
      "name": "Andres",
    },
    {
      "id": 11191,
      "name": "Nautix",
    },
    {
      "id": 11192,
      "name": "Torivallavalitsus",
    },
    {
      "id": 11193,
      "name": "Ninne",
    },
    {
      "id": 11194,
      "name": "Urmas Varris",
    },
    {
      "id": 11195,
      "name": "WindWalker",
    },
    {
      "id": 11196,
      "name": "Via Grupp OU (Командор)",
    },
    {
      "id": 11201,
      "name": "Tõnis",
    },
    {
      "id": 11202,
      "name": "Jaanus",
    },
    {
      "id": 11203,
      "name": "Meister Arle OU",
    },
    {
      "id": 11204,
      "name": "Prime Marine Service OU",
    },
    {
      "id": 11205,
      "name": "Viking",
    },
    {
      "id": 11206,
      "name": "Saku",
    },
    {
      "id": 11207,
      "name": "SineneLind",
    },
    {
      "id": 11208,
      "name": "Alex Tido",
    },
    {
      "id": 11209,
      "name": "AZZURRA",
    },
    {
      "id": 11210,
      "name": "квас на пиво",
    },
    {
      "id": 11211,
      "name": "MIA (Tallin)",
    },
    {
      "id": 11212,
      "name": "MINNI",
    },
    {
      "id": 11213,
      "name": "Jonпgert OU",
    },
    {
      "id": 11214,
      "name": "MIA Pärnu",
    },
    {
      "id": 11217,
      "name": "Tauno Teluk",
    },
    {
      "id": 11219,
      "name": "Капитан",
    },
    {
      "id": 11220,
      "name": "Johannes  Atlandi Võlg",
    },
    {
      "id": 11221,
      "name": "Nimi",
    },
    {
      "id": 11222,
      "name": "Viahooldus ОÜ",
    },
    {
      "id": 11223,
      "name": "Kaupmees",
    },
    {
      "id": 11224,
      "name": "Reino",
    },
    {
      "id": 11225,
      "name": "Rubit OU",
    },
    {
      "id": 11226,
      "name": "Aleksey  Rusetsky",
    },
    {
      "id": 11227,
      "name": "Corinthian Events OÜ",
    },
    {
      "id": 11228,
      "name": "Ronstan DK",
    },
    {
      "id": 11229,
      "name": "Narva-Joesuu PurjeSpoordiKool",
    },
    {
      "id": 11230,
      "name": "Saaremaa Meri Spordiselts",
    },
    {
      "id": 11231,
      "name": "Tallux Orto",
    },
    {
      "id": 11232,
      "name": "Uku Kuusk",
    },
    {
      "id": 11234,
      "name": "Bainbridge Inc (UK)",
    },
    {
      "id": 11235,
      "name": "BetaMarine (UK)",
    },
    {
      "id": 11236,
      "name": "Change Consulting OÜ",
    },
    {
      "id": 11237,
      "name": "Rait Maru",
    },
    {
      "id": 11239,
      "name": "Žaklar OU",
    },
    {
      "id": 11240,
      "name": "Comar Systems Limited",
    },
    {
      "id": 11241,
      "name": "Kaidor",
    },
    {
      "id": 11242,
      "name": "MADEN",
    },
    {
      "id": 11243,
      "name": "Ropebox Eesti OÜ",
    },
    {
      "id": 11244,
      "name": "Muri",
    },
    {
      "id": 11245,
      "name": "Carrestoration OU",
    },
    {
      "id": 11246,
      "name": "Siren Sailing OU",
    },
    {
      "id": 11247,
      "name": "Episood OU",
    },
    {
      "id": 11248,
      "name": "Akster Grupp OU",
    },
    {
      "id": 11249,
      "name": "Alter",
    },
    {
      "id": 11250,
      "name": "Ronstan AUS",
    },
    {
      "id": 11251,
      "name": "Innon Finance OU",
    },
    {
      "id": 11252,
      "name": "Lettore OÜ",
    },
    {
      "id": 11253,
      "name": "Navideep Services OÜ",
    },
    {
      "id": 11254,
      "name": "Athaman (Polska)",
    },
    {
      "id": 11255,
      "name": "Lõbusõit OÜ",
    },
    {
      "id": 11256,
      "name": "Aleksei Mikhalev",
    },
    {
      "id": 11257,
      "name": "Ropebox OÜ",
    },
    {
      "id": 11258,
      "name": "Kaubakeskus OÜ",
    },
    {
      "id": 11259,
      "name": "Rein Ottoson OÜ",
    },
    {
      "id": 11260,
      "name": "Robert",
    },
    {
      "id": 11261,
      "name": "Merema OÜ",
    },
    {
      "id": 11262,
      "name": "Alexander Rusetsky",
    },
    {
      "id": 11263,
      "name": "Natalja Rusetskaya",
    },
    {
      "id": 11264,
      "name": "Jahtklubi Dago",
    },
    {
      "id": 11265,
      "name": "Elev Luik",
    },
    {
      "id": 11266,
      "name": "_ЗХ2023",
    },
    {
      "id": 11267,
      "name": "Vsevolod pyramidaprint",
    },
    {
      "id": 11268,
      "name": "Emerald",
    },
    {
      "id": 11270,
      "name": "Strand Sailing Management",
    },
    {
      "id": 11271,
      "name": "Tuule",
    },
    {
      "id": 11272,
      "name": "Meelis Kojamess (Баба Яга)",
    },
    {
      "id": 11273,
      "name": "Andrea FE64",
    },
    {
      "id": 11274,
      "name": "Reilut OÜ",
    },
    {
      "id": 11275,
      "name": "Pärnu cruises",
    },
    {
      "id": 11280,
      "name": "Madis Parts",
    },
    {
      "id": 11281,
      "name": "Tera ehitus OU",
    },
    {
      "id": 11282,
      "name": "Korkvender OU",
    },
    {
      "id": 11291,
      "name": "STRAND PROJEKT OÜ",
    },
    {
      "id": 11292,
      "name": "RTR Tehnika",
    },
    {
      "id": 11293,
      "name": "Eprodukt  OÜ",
    },
    {
      "id": 11294,
      "name": "Madis Parts",
    },
    {
      "id": 11295,
      "name": "SeaKayaking",
    },
    {
      "id": 11296,
      "name": "8 Motor Oy",
    },
    {
      "id": 11297,
      "name": "SKELTMER OÜ",
    },
    {
      "id": 11298,
      "name": "Jäälind",
    },
    {
      "id": 11299,
      "name": "MattChem (France)",
    },
    {
      "id": 11300,
      "name": "Car Restoration OU",
    },
    {
      "id": 11301,
      "name": "Valgeranna Seikluspark OÜ",
    },
    {
      "id": 11302,
      "name": "Kaif OÜ",
    },
    {
      "id": 11303,
      "name": "Julechka",
    },
    {
      "id": 11304,
      "name": "Longboat",
    },
    {
      "id": 11305,
      "name": "SAILING MASSIVE  OÜ",
    },
    {
      "id": 11306,
      "name": "AB Technology",
    },
    {
      "id": 11307,
      "name": "Norman Saarso",
    },
    {
      "id": 11308,
      "name": "Tuisu OÜ",
    },
    {
      "id": 11309,
      "name": "Fairaway",
    },
    {
      "id": 11311,
      "name": "Alibi OÜ",
    },
    {
      "id": 11312,
      "name": "Dirham motorpaad",
    },
    {
      "id": 11313,
      "name": "Niined group",
    },
    {
      "id": 11314,
      "name": "Magna Fide",
    },
    {
      "id": 11315,
      "name": "Корячкин",
    },
    {
      "id": 11317,
      "name": "Konstrado",
    },
    {
      "id": 11318,
      "name": "Vektom OÜ",
    },
    {
      "id": 11319,
      "name": "Kalev Jahtklubi",
    },
    {
      "id": 11320,
      "name": "Marta Vintage OÜ",
    },
  ],
  "orders": [
    {
      "allSum": 624.2103,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-09-25T12:02:00.000Z",
      "id": 205282,
      "isPaid": true,
      "paidSum": 624.2103,
      "payType": "Visa, Нал",
      "positionCount": 3,
      "productsCount": -15,
      "productsSum": 624.2103,
      "status": "продажа",
    },
    {
      "allSum": 882.8462,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-09-03T11:52:00.000Z",
      "id": 205224,
      "isPaid": true,
      "paidSum": 882.8462,
      "payType": "Нал",
      "positionCount": 8,
      "productsCount": -20,
      "productsSum": 882.8462,
      "status": "продажа",
    },
    {
      "allSum": 5638.2805,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-09-03T11:54:00.000Z",
      "id": 205222,
      "isPaid": true,
      "paidSum": 5638.2805,
      "payType": "Нал",
      "positionCount": 36,
      "productsCount": -718,
      "productsSum": 5638.2805,
      "status": "продажа",
    },
    {
      "allSum": 1587.0671,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-08-12T18:28:00.000Z",
      "id": 205142,
      "isPaid": true,
      "paidSum": 1587.0671,
      "payType": "Visa, Нал",
      "positionCount": 14,
      "productsCount": -598,
      "productsSum": 1587.0671,
      "status": "продажа",
    },
    {
      "allSum": 110.656,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-07-24T13:27:00.000Z",
      "id": 205018,
      "isPaid": true,
      "paidSum": 110.656,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 110.656,
      "status": "продажа",
    },
    {
      "allSum": 1387.3785,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-07-24T12:58:00.000Z",
      "id": 205012,
      "isPaid": true,
      "paidSum": 1387.3785,
      "payType": "Нал",
      "positionCount": 15,
      "productsCount": -145,
      "productsSum": 1387.3785,
      "status": "продажа",
    },
    {
      "allSum": 1926.8713,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-07-01T15:39:00.000Z",
      "id": 204845,
      "isPaid": true,
      "paidSum": 1926.8713,
      "payType": "Нал",
      "positionCount": 13,
      "productsCount": -228,
      "productsSum": 1926.8713,
      "status": "продажа",
    },
    {
      "allSum": 1186.402,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-07-24T13:11:00.000Z",
      "id": 204826,
      "isPaid": true,
      "paidSum": 1186.402,
      "payType": "Нал",
      "positionCount": 10,
      "productsCount": -102,
      "productsSum": 1186.402,
      "status": "продажа",
    },
    {
      "allSum": 1240.7201,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-08-12T17:58:00.000Z",
      "id": 204800,
      "isPaid": true,
      "paidSum": 1240.7201,
      "payType": "Нал",
      "positionCount": 5,
      "productsCount": -15,
      "productsSum": 1240.7201,
      "status": "продажа",
    },
    {
      "allSum": 998.89,
      "clientName": "Aleksey  Rusetsky",
      "date": "2022-07-01T15:32:00.000Z",
      "id": 204798,
      "isPaid": true,
      "paidSum": 998.89,
      "payType": "Нал",
      "positionCount": 1,
      "productsCount": -2,
      "productsSum": 998.89,
      "status": "продажа",
    },
  ],
  "pagesCount": 3,
  "rowsCount": 21,
  "totalAllSum": 22751.0654,
  "totalPaidSum": 22850.7162,
  "totalProductSum": 22751.0654,
}
`;

exports[`Client Filter Get Products client: 11226 1`] = `
{
  "pagesCount": 15,
  "products": [
    {
      "avgPrice": 231.0589,
      "id": 58224,
      "maxPrice": 231.0589,
      "minPrice": 231.0589,
      "name": "DRIVEPLATE SAE6.5 -HURTH / PRM 22-B-60",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 231.0589,
    },
    {
      "avgPrice": 17.6811,
      "id": 58223,
      "maxPrice": 17.6811,
      "minPrice": 17.6811,
      "name": "Pa PRESSURE CAP TWIN SEAL",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 35.3622,
    },
    {
      "avgPrice": 74.4768,
      "id": 58201,
      "maxPrice": 74.4768,
      "minPrice": 74.4768,
      "name": "Закрутка геннакера серия 60. верхний шарнир",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 148.9536,
    },
    {
      "avgPrice": 121.3006,
      "id": 58200,
      "maxPrice": 121.3006,
      "minPrice": 121.3006,
      "name": "Закрутка для геннакера серия 60. нижняя часть",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 242.6012,
    },
    {
      "avgPrice": 12.0499,
      "id": 58178,
      "maxPrice": 12.0499,
      "minPrice": 12.0499,
      "name": "AIRX-900N Red 152cm",
      "ordersCount": 1,
      "totalCount": -20,
      "totalSum": 240.998,
    },
    {
      "avgPrice": 8.5654,
      "id": 58177,
      "maxPrice": 8.5654,
      "minPrice": 8.5654,
      "name": "Sail Repair Patches Assorted Sizes Heavy Duty White",
      "ordersCount": 1,
      "totalCount": -4,
      "totalSum": 34.2616,
    },
    {
      "avgPrice": 28.7524,
      "id": 58160,
      "maxPrice": 32.6411,
      "minPrice": 24.8637,
      "name": "Rope bag Grey PVC|Mesh wide 400x250x200mm",
      "ordersCount": 2,
      "totalCount": -8,
      "totalSum": 237.7966,
    },
    {
      "avgPrice": 590.1167,
      "id": 58159,
      "maxPrice": 590.1167,
      "minPrice": 590.1167,
      "name": "SeaProp60 MEMBRANE (INC WATER SENSOR TAPPING) 05-2012176",
      "ordersCount": 1,
      "totalCount": -1,
      "totalSum": 590.1167,
    },
    {
      "avgPrice": 154.2351,
      "id": 58158,
      "maxPrice": 154.2351,
      "minPrice": 154.2351,
      "name": "OIL SEAL, ZF SD10, 3307.301.016.01",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 308.4702,
    },
    {
      "avgPrice": 499.445,
      "id": 58157,
      "maxPrice": 499.445,
      "minPrice": 499.445,
      "name": "R400N Network AIS Receiver with Ethernet",
      "ordersCount": 1,
      "totalCount": -2,
      "totalSum": 998.89,
    },
  ],
  "rowsCount": 143,
}
`;

exports[`Client Filter Get Sales client: 11243 1`] = `
{
  "clients": [
    {
      "id": null,
      "name": null,
    },
    {
      "id": 245,
      "name": "Щербаков Антон Васильевич",
    },
    {
      "id": 627,
      "name": "Sailtech Oy (Финляндия)",
    },
    {
      "id": 676,
      "name": "Maritim-Finland",
    },
    {
      "id": 940,
      "name": "web",
    },
    {
      "id": 1108,
      "name": "TREM",
    },
    {
      "id": 1702,
      "name": "MarineTech (Germany)",
    },
    {
      "id": 2766,
      "name": "Optiparts",
    },
    {
      "id": 2773,
      "name": "Windex (Sweden)",
    },
    {
      "id": 9901,
      "name": "Anton Scherbakov",
    },
    {
      "id": 11182,
      "name": "_jamuuk",
    },
    {
      "id": 11183,
      "name": "PJK",
    },
    {
      "id": 11185,
      "name": "Regate (LV)",
    },
    {
      "id": 11186,
      "name": "TommyyLyy",
    },
    {
      "id": 11187,
      "name": "Pro Marine Trade",
    },
    {
      "id": 11188,
      "name": "Татьяна Каверина (Armastus)",
    },
    {
      "id": 11189,
      "name": "NORA LII SOPRADE SELTS",
    },
    {
      "id": 11190,
      "name": "Andres",
    },
    {
      "id": 11191,
      "name": "Nautix",
    },
    {
      "id": 11192,
      "name": "Torivallavalitsus",
    },
    {
      "id": 11193,
      "name": "Ninne",
    },
    {
      "id": 11194,
      "name": "Urmas Varris",
    },
    {
      "id": 11195,
      "name": "WindWalker",
    },
    {
      "id": 11196,
      "name": "Via Grupp OU (Командор)",
    },
    {
      "id": 11201,
      "name": "Tõnis",
    },
    {
      "id": 11202,
      "name": "Jaanus",
    },
    {
      "id": 11203,
      "name": "Meister Arle OU",
    },
    {
      "id": 11204,
      "name": "Prime Marine Service OU",
    },
    {
      "id": 11205,
      "name": "Viking",
    },
    {
      "id": 11206,
      "name": "Saku",
    },
    {
      "id": 11207,
      "name": "SineneLind",
    },
    {
      "id": 11208,
      "name": "Alex Tido",
    },
    {
      "id": 11209,
      "name": "AZZURRA",
    },
    {
      "id": 11210,
      "name": "квас на пиво",
    },
    {
      "id": 11211,
      "name": "MIA (Tallin)",
    },
    {
      "id": 11212,
      "name": "MINNI",
    },
    {
      "id": 11213,
      "name": "Jonпgert OU",
    },
    {
      "id": 11214,
      "name": "MIA Pärnu",
    },
    {
      "id": 11217,
      "name": "Tauno Teluk",
    },
    {
      "id": 11219,
      "name": "Капитан",
    },
    {
      "id": 11220,
      "name": "Johannes  Atlandi Võlg",
    },
    {
      "id": 11221,
      "name": "Nimi",
    },
    {
      "id": 11222,
      "name": "Viahooldus ОÜ",
    },
    {
      "id": 11223,
      "name": "Kaupmees",
    },
    {
      "id": 11224,
      "name": "Reino",
    },
    {
      "id": 11225,
      "name": "Rubit OU",
    },
    {
      "id": 11226,
      "name": "Aleksey  Rusetsky",
    },
    {
      "id": 11227,
      "name": "Corinthian Events OÜ",
    },
    {
      "id": 11228,
      "name": "Ronstan DK",
    },
    {
      "id": 11229,
      "name": "Narva-Joesuu PurjeSpoordiKool",
    },
    {
      "id": 11230,
      "name": "Saaremaa Meri Spordiselts",
    },
    {
      "id": 11231,
      "name": "Tallux Orto",
    },
    {
      "id": 11232,
      "name": "Uku Kuusk",
    },
    {
      "id": 11234,
      "name": "Bainbridge Inc (UK)",
    },
    {
      "id": 11235,
      "name": "BetaMarine (UK)",
    },
    {
      "id": 11236,
      "name": "Change Consulting OÜ",
    },
    {
      "id": 11237,
      "name": "Rait Maru",
    },
    {
      "id": 11239,
      "name": "Žaklar OU",
    },
    {
      "id": 11240,
      "name": "Comar Systems Limited",
    },
    {
      "id": 11241,
      "name": "Kaidor",
    },
    {
      "id": 11242,
      "name": "MADEN",
    },
    {
      "id": 11243,
      "name": "Ropebox Eesti OÜ",
    },
    {
      "id": 11244,
      "name": "Muri",
    },
    {
      "id": 11245,
      "name": "Carrestoration OU",
    },
    {
      "id": 11246,
      "name": "Siren Sailing OU",
    },
    {
      "id": 11247,
      "name": "Episood OU",
    },
    {
      "id": 11248,
      "name": "Akster Grupp OU",
    },
    {
      "id": 11249,
      "name": "Alter",
    },
    {
      "id": 11250,
      "name": "Ronstan AUS",
    },
    {
      "id": 11251,
      "name": "Innon Finance OU",
    },
    {
      "id": 11252,
      "name": "Lettore OÜ",
    },
    {
      "id": 11253,
      "name": "Navideep Services OÜ",
    },
    {
      "id": 11254,
      "name": "Athaman (Polska)",
    },
    {
      "id": 11255,
      "name": "Lõbusõit OÜ",
    },
    {
      "id": 11256,
      "name": "Aleksei Mikhalev",
    },
    {
      "id": 11257,
      "name": "Ropebox OÜ",
    },
    {
      "id": 11258,
      "name": "Kaubakeskus OÜ",
    },
    {
      "id": 11259,
      "name": "Rein Ottoson OÜ",
    },
    {
      "id": 11260,
      "name": "Robert",
    },
    {
      "id": 11261,
      "name": "Merema OÜ",
    },
    {
      "id": 11262,
      "name": "Alexander Rusetsky",
    },
    {
      "id": 11263,
      "name": "Natalja Rusetskaya",
    },
    {
      "id": 11264,
      "name": "Jahtklubi Dago",
    },
    {
      "id": 11265,
      "name": "Elev Luik",
    },
    {
      "id": 11266,
      "name": "_ЗХ2023",
    },
    {
      "id": 11267,
      "name": "Vsevolod pyramidaprint",
    },
    {
      "id": 11268,
      "name": "Emerald",
    },
    {
      "id": 11270,
      "name": "Strand Sailing Management",
    },
    {
      "id": 11271,
      "name": "Tuule",
    },
    {
      "id": 11272,
      "name": "Meelis Kojamess (Баба Яга)",
    },
    {
      "id": 11273,
      "name": "Andrea FE64",
    },
    {
      "id": 11274,
      "name": "Reilut OÜ",
    },
    {
      "id": 11275,
      "name": "Pärnu cruises",
    },
    {
      "id": 11280,
      "name": "Madis Parts",
    },
    {
      "id": 11281,
      "name": "Tera ehitus OU",
    },
    {
      "id": 11282,
      "name": "Korkvender OU",
    },
    {
      "id": 11291,
      "name": "STRAND PROJEKT OÜ",
    },
    {
      "id": 11292,
      "name": "RTR Tehnika",
    },
    {
      "id": 11293,
      "name": "Eprodukt  OÜ",
    },
    {
      "id": 11294,
      "name": "Madis Parts",
    },
    {
      "id": 11295,
      "name": "SeaKayaking",
    },
    {
      "id": 11296,
      "name": "8 Motor Oy",
    },
    {
      "id": 11297,
      "name": "SKELTMER OÜ",
    },
    {
      "id": 11298,
      "name": "Jäälind",
    },
    {
      "id": 11299,
      "name": "MattChem (France)",
    },
    {
      "id": 11300,
      "name": "Car Restoration OU",
    },
    {
      "id": 11301,
      "name": "Valgeranna Seikluspark OÜ",
    },
    {
      "id": 11302,
      "name": "Kaif OÜ",
    },
    {
      "id": 11303,
      "name": "Julechka",
    },
    {
      "id": 11304,
      "name": "Longboat",
    },
    {
      "id": 11305,
      "name": "SAILING MASSIVE  OÜ",
    },
    {
      "id": 11306,
      "name": "AB Technology",
    },
    {
      "id": 11307,
      "name": "Norman Saarso",
    },
    {
      "id": 11308,
      "name": "Tuisu OÜ",
    },
    {
      "id": 11309,
      "name": "Fairaway",
    },
    {
      "id": 11311,
      "name": "Alibi OÜ",
    },
    {
      "id": 11312,
      "name": "Dirham motorpaad",
    },
    {
      "id": 11313,
      "name": "Niined group",
    },
    {
      "id": 11314,
      "name": "Magna Fide",
    },
    {
      "id": 11315,
      "name": "Корячкин",
    },
    {
      "id": 11317,
      "name": "Konstrado",
    },
    {
      "id": 11318,
      "name": "Vektom OÜ",
    },
    {
      "id": 11319,
      "name": "Kalev Jahtklubi",
    },
    {
      "id": 11320,
      "name": "Marta Vintage OÜ",
    },
  ],
  "orders": [
    {
      "allSum": -2799.5089,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-08-10T15:38:00.000Z",
      "id": 205162,
      "isPaid": false,
      "paidSum": -2799.51,
      "payType": "Б/н",
      "positionCount": 30,
      "productsCount": 4074,
      "productsSum": -2799.5089,
      "status": "покупка",
    },
    {
      "allSum": -303.4701,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-05-26T14:41:00.000Z",
      "id": 205159,
      "isPaid": false,
      "paidSum": -303.47,
      "payType": "Б/н",
      "positionCount": 6,
      "productsCount": 12,
      "productsSum": -303.4701,
      "status": "покупка",
    },
    {
      "allSum": -4676.102,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-05-06T14:15:00.000Z",
      "id": 205158,
      "isPaid": false,
      "paidSum": null,
      "payType": null,
      "positionCount": 77,
      "productsCount": 236,
      "productsSum": -4676.102,
      "status": "покупка",
    },
    {
      "allSum": -708.9206,
      "clientName": "Ropebox Eesti OÜ",
      "date": "2022-07-01T12:27:00.000Z",
      "id": 205155,
      "isPaid": false,
      "paidSum": -708.92,
      "payType": "Б/н",
      "positionCount": 9,
      "productsCount": 29,
      "productsSum": -708.9206,
      "status": "покупка",
    },
  ],
  "pagesCount": 1,
  "rowsCount": 4,
  "totalAllSum": -8488.0016,
  "totalPaidSum": -3811.9,
  "totalProductSum": -8488.0016,
}
`;

exports[`Client Filter Get Products client: 11243 1`] = `
{
  "pagesCount": 11,
  "products": [
    {
      "avgPrice": 31.0796,
      "id": 58160,
      "maxPrice": 31.0796,
      "minPrice": 31.0796,
      "name": "Rope bag Grey PVC|Mesh wide 400x250x200mm",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -62.1592,
    },
    {
      "avgPrice": 48.6071,
      "id": 58113,
      "maxPrice": 48.6071,
      "minPrice": 48.6071,
      "name": "Часы ClearStart - Sailing Watch, синие",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -48.6071,
    },
    {
      "avgPrice": 48.6071,
      "id": 58112,
      "maxPrice": 48.6071,
      "minPrice": 48.6071,
      "name": "Часы ClearStart - Sailing Watch, чер-крас-бел цвет",
      "ordersCount": 1,
      "totalCount": 1,
      "totalSum": -48.6071,
    },
    {
      "avgPrice": 25.7226,
      "id": 55550,
      "maxPrice": 25.7226,
      "minPrice": 25.7226,
      "name": "Перчатки черные, длинные пальцы XL",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -51.4452,
    },
    {
      "avgPrice": 25.9573,
      "id": 55549,
      "maxPrice": 26.192,
      "minPrice": 25.7226,
      "name": "Перчатки черные, длинные пальцы S",
      "ordersCount": 2,
      "totalCount": 6,
      "totalSum": -156.2132,
    },
    {
      "avgPrice": 25.9573,
      "id": 55548,
      "maxPrice": 26.192,
      "minPrice": 25.7226,
      "name": "Перчатки черные, длинные пальцы M",
      "ordersCount": 2,
      "totalCount": 8,
      "totalSum": -207.6584,
    },
    {
      "avgPrice": 25.7226,
      "id": 55337,
      "maxPrice": 25.7226,
      "minPrice": 25.7226,
      "name": "Перчатки черные, L",
      "ordersCount": 1,
      "totalCount": 4,
      "totalSum": -102.8904,
    },
    {
      "avgPrice": 21.8716,
      "id": 54508,
      "maxPrice": 21.8716,
      "minPrice": 21.8716,
      "name": "Блок двойной вертлюжный",
      "ordersCount": 1,
      "totalCount": 2,
      "totalSum": -43.7432,
    },
    {
      "avgPrice": 7.6204,
      "id": 53740,
      "maxPrice": 7.6204,
      "minPrice": 7.6204,
      "name": "Лента для ремонта RED 50mm x 10m (замена J601RD)",
      "ordersCount": 1,
      "totalCount": 10,
      "totalSum": -76.204,
    },
    {
      "avgPrice": 12.07,
      "id": 51174,
      "maxPrice": 12.07,
      "minPrice": 12.07,
      "name": "Стопор кулачковый Small красная вставка",
      "ordersCount": 1,
      "totalCount": 4,
      "totalSum": -48.28,
    },
  ],
  "rowsCount": 102,
}
`;
